subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -310,7 +310,7 @@ class RealSubscriptionsManager @Inject constructor(
             if (isUserAuthenticated()) {
                 getSubscriptionDataFromToken(authRepository.tokens().accessToken!!)
             } else {
-                Failure("Subscription data not found")
+                Success(email = null, externalId = "", entitlements = emptyList())
```

Comment:
FYI this is needed so when the user signs out, rather than returning a failure we return an empty subscription so the state in the settings screen can change. Failure will now only happen when there's an actual failure.